sentry-android-core/src/main/java/io/sentry/android/core/CurrentActivityHolder.java

```diff
@@ -0,0 +1,39 @@
+package io.sentry.android.core;
+
+import android.app.Activity;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import java.lang.ref.WeakReference;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+@ApiStatus.Internal
+public class CurrentActivityHolder {
+
+  private static final @NotNull CurrentActivityHolder instance = new CurrentActivityHolder();
+
```

Comment:
`m`: can we also add a private constructor here so it's not possible to instantiate the `CurrentActivityHolder` and we are always forced to use the singleton instance?
```suggestion

    private CurrentActivityHolder() { }
```